src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunction.cs

```diff
@@ -72,4 +45,10 @@ public abstract class AIFunction : AITool
     protected abstract ValueTask<object?> InvokeCoreAsync(
         AIFunctionArguments arguments,
         CancellationToken cancellationToken);
+
+    /// <summary>Creates a <see cref="AIFunctionDefinition"/> representation of this <see cref="AIFunction"/> that can't be invoked.</summary>
+    /// <returns>The created instance.</returns>
+    public AIFunctionDefinition AsDefinitionOnly() => new NonInvocableAIFunction(this);
```

Comment:
Maybe I'll spot the reasoning later in the PR review, but I'm not certain at this point what the use case for this is.

Reply:
Primarily to enable skipping any function invoking middleware in the pipeline (I need to make a few tweaks to FunctionInvokingChatClient for this to actually work). I can put a non-AIFunction AIFunctionDefinition into ChatOptions.Tools, and then FunctionCallContent for it won't be automatically handled by the FICC but will instead pass through.